### PR TITLE
[Model] Add random_start option for Farthest Point Sampler

### PR DESCRIPTION
**Pull request by lygztq**
## Description
Add `random_start` option for FPS. 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/geometry/mxnet/fps.py [hunk 1]
```diff
@@ -24,13 +24,16 @@ def __init__(self, npoints):
         super(FarthestPointSampler, self).__init__()
         self.npoints = npoints
 
-    def forward(self, pos):
+    def forward(self, pos, random_start=True):
```
Comment (hetong007) [inline python/dgl/geometry/mxnet/fps.py:27]: Can we change it to `start_idx`, which appoint the index of the starting point if is not `None`?

File: python/dgl/geometry/mxnet/fps.py [hunk 2]
```diff
@@ -44,10 +46,13 @@ def forward(self, pos, random_start=True):
         B, N, C = pos.shape
         pos = pos.reshape(-1, C)
         dist = nd.zeros((B * N), dtype=pos.dtype, ctx=ctx)
-        if random_start:
+        if start_idx is None:
             start_idx = nd.random.randint(0, N - 1, (B, ), dtype=np.int, ctx=ctx)
         else:
-            start_idx = nd.zeros((B, ), dtype=np.int, ctx=ctx)
+            if start_idx > N:
```
Comment (hetong007) [inline python/dgl/geometry/mxnet/fps.py:52]: Let it be `start_idx >= N and start_idx < 0`

File: python/dgl/geometry/mxnet/fps.py [hunk 3]
```diff
@@ -44,10 +46,13 @@ def forward(self, pos, random_start=True):
...
+                raise DGLError("Invalid start_idx, expected index <= {}, got {}".format(
```
Comment (hetong007) [inline python/dgl/geometry/mxnet/fps.py:53]: and the error message needs to change accordingly.

**Review by hetong007 — COMMENTED** [review]
Two comments, otherwise LGTM.

File: python/dgl/geometry/mxnet/fps.py (hunk 3 above)
Comment (lygztq) [reply python/dgl/geometry/mxnet/fps.py:53]: solved

Comment (hetong007) [reply python/dgl/geometry/mxnet/fps.py:53]: I mean the message needs to be something like `expected 0 <= index < {}`.

Comment (lygztq) [reply python/dgl/geometry/mxnet/fps.py:53]: oh, my fault

**Review by hetong007 — APPROVED** [review]

**Pull request merged**